src/com/palmergames/bukkit/config/ConfigNodes.java

```diff
@@ -281,6 +281,13 @@ public enum ConfigNodes {
 			"# at the cost of more work setting up. Also, extremely small values will render the caching done useless.",
 			"# Each cell is (town_block_size * town_block_size * height-of-the-world) in size, with height-of-the-world",
 			"# being from the bottom to the top of the build-able world."),
+	TOWN_MIN_ADJACENT_BLOCKS(
+			"town.min_adjacent_blocks",
+			"-1",
+			"",
+			"# The minimum adjacent town blocks required to expand.",
+			"# This can prevent long lines and snake-like patterns.",
+			"# Set to -1 to disable. Set to 3 to force wider expansions of towns."),

```

Comment:
I don't believe this feature will work if 2 or 3 is set. You could never claim beyond the homeblock if 2 or greater adjacent townblocks are required.

Even 2 I think would prevent someone from claiming more land after an initial 3x3 claim.

Reply:
I took that into consideration, so the town size has to be greater than minAdjacentBlocks for this expansion constraint to be taken into consideration.

Bare in mind that I have adjusted the function that detects nearby town blocks to also look at the diagonal coordinates.

Setting it to 1 doesn't accomplish anything that a border check doesn't already do.

Setting it to 2 allows fatter line expansion at a 1x2 rate (doubling the amount of nearby town blocks).

Setting it to 3 does the same... 1x3 expansion is technically possible.

Setting it above 3 would require more advanced logic to detect further away town blocks.

Reply:
I did not catch the part where town size was being considered. I've done some testing on pen & paper and this will work when 3 is set, claiming progression is a bit less free but I suppose that's what the admin would want in this scenario.